### PR TITLE
Implement all methods in accordance with the description

### PR DESCRIPTION
**Pull request by JewelSam**

### CODE REVIEW

File: lib/tram/policy/errors.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    class Errors
+      include Enumerable
+      extend Forwardable
+
+      def_delegators :@errors, :each, :size, :empty?
+      attr_reader :errors
+
+      def initialize(policy)
+        @policy = policy
+        @errors = []
```
Comment (nepalez) [inline lib/tram/policy/errors.rb]: It's better to use `set` here -- for two reasons (1) errors aren't ordered, (2) they should be unique

PS. That's why we needed `Error#==`

File: lib/tram/policy/error.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      # Say you have class Article::ReadinessPolicy < Tram::Policy; end
+      # and you wanted the translation for the :blank error message for
+      # the title attribute, it looks for this translation: article/readiness_policy.empty
+      def generate_message(message)
+        if message.is_a?(String)
+          message
+        elsif message.is_a?(Symbol)
+          I18n.t(message, @tags.merge(scope: i18n_scope, default: "Error translation for missed text"))
+        else
+          raise ArgumentError.new("Only strings or symbols are allowed")
```
Comment (JewelSam) [inline lib/tram/policy/error.rb]: fixed

File: lib/tram/policy/error.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        end
+      end
+
+      # Return an array of missed translations to every available locale
+      #
+      #   Tram::Policy::Error.new article_policy, :empty_title, field: 'title'
+      #   error.missed_translations # => ['en.article.empty_title', 'ru.article.empty_title']
+      def missed_translations
```
Comment (JewelSam) [inline lib/tram/policy/error.rb]: It's not only test-related stuff. Tram::Policy#invalid_at? returns missed translations. This requirement was in Readme: 

> Notice that RSpec matcher be_invalid_at checks at once errors and provides full description for the essence of the failure.

Or maybe I misunderstood something?

File: lib/tram/policy/error.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      #   error = Tram::Policy::Error.new(policy, "Title is empty", field: "title", level: "error")
+      #   error.to_h # => {message: message, field: 'title', level: 'error'}
+      def to_h
+        tags.merge(message: message)
+      end
+
+      # Checks whether an error is equal to another one
+      # Errors are equal if policies, tags and messages are equal
+      def ==(error)
+        policy == error.policy && to_h == error.to_h
```
Comment (JewelSam) [inline lib/tram/policy/error.rb]: fixed

File: lib/tram/policy/error.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      end
+
+      # Undefined methods treated as tags
+      def method_missing(m, *args, &block)
+        if tags.has_key?(m.to_sym)
```
Comment (JewelSam) [inline lib/tram/policy/error.rb]: fixed

File: lib/tram/policy/error.rb [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+          tags[m.to_sym]
+        else
+          super
+        end
+      end
+
+      private
+        def i18n_scope
+          @policy.class.name.underscore
```
Comment (JewelSam) [inline lib/tram/policy/error.rb]: I added #underscore method to lib/core_ext/string.rb (almost copy-paste :)), so it is not rails-dependent

File: lib/tram/policy/error.rb (hunk 3 above)
Comment (nepalez) [inline lib/tram/policy/error.rb]: I meant https://www.relishapp.com/rspec/rspec-expectations/v/3-5/docs/custom-matchers
(I warmly recommend you reading the whole RSpec documentation on relish -- it's worth it)

File: lib/tram/policy/error.rb (hunk 6 above)
Comment (nepalez) [inline lib/tram/policy/error.rb]: Ahh, I see... this way is far too railsy

I would treat any monkey patching outside of a library as an immemorial evil (the one that made `ActiveSupport` such an unbearable dump of heap!), and would recommend keeping aside of it for personal safety ;)

In some cases lexical scoping may be ok (https://ruby-doc.org/core-2.1.1/doc/syntax/refinements_rdoc.html, http://yehudakatz.com/2010/11/30/ruby-2-0-refinements-in-practice/). Tbh, on my taste this style of coding still provide too clumsy, non-readable code with unexpected behaviour.

You could build much cleaner code by extracting inflector into utility module with one or two methods (functional approach of sort).

PS. 1.day.ago... 1.bottle.of.beer OMG!

**Pull request merged**